ddtrace/tracer.py

```diff
@@ -88,6 +88,51 @@
 AnyCallable = TypeVar("AnyCallable", bound=Callable)
 
 
+def _default_span_processors_factory(
+    trace_filters,  # type: List[TraceFilter]
+    trace_writer,  # type: TraceWriter
+    partial_flush_enabled,  # type: bool
+    partial_flush_min_spans,  # type: int
+    appsec_enabled,  # type: bool
+):
+    # type: (...) -> List[SpanProcessor]
+    """Construct the default list of span processors to use."""
+    trace_processors = []  # type: List[TraceProcessor]
+    trace_processors += [TraceTagsProcessor()]
+    trace_processors += [TraceSamplingProcessor()]
+    trace_processors += [TraceTopLevelSpanProcessor()]
+    trace_processors += trace_filters
+
+    span_processors = []  # type: List[SpanProcessor]
+
+    if appsec_enabled:
```

Comment:
```suggestion
    if self._appsec_enabled:
```

Reply:
there is no self reference here 🤔

Reply:
I missed that. Would it make sense to move `` _default_span_processors_factory`` into the Tracer object?

Reply:
Yeah I opted to keep it separate so that it isn't dependent on any implicit tracer state so that we avoid ordering bugs.